File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        count = [0] * 26

        for c in s:
            count[ord(c) - ord('a')] += 1

        for c in t:
            count[ord(c) - ord('a')] -= 1

        for c in count:
            if c != 0:
                return False

        return True

File: 0242-valid-anagram/0242-valid-anagram.ts
function isAnagram(s: string, t: string): boolean {
    if (s.length !== t.length) {
        return false;
    }

    const count: number[] = new Array(26).fill(0);

    for (const c of s) {
        count[c.charCodeAt(0) - 'a'.charCodeAt(0)]++;
    }

    for (const c of t) {
        count[c.charCodeAt(0) - 'a'.charCodeAt(0)]--;
    }

    for (const val of count) {
        if (val !== 0) {
            return false;
        }
    }

    return true;
}
